lib/project_types/theme/models/specification_handlers/theme.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Theme
+  module Models
+    module SpecificationHandlers
+      class Theme
+        REQUIRED_FOLDERS = %w(config layout sections templates).map { |folder| "#{folder}/" }
+
+        def initialize(root)
+          Dir.chdir(root) do
```

Comment:
Nitpick: Would it work if instead of choosing a directory through `Dir.chdir`, we do:
```rb
self.folders = Dir[File.join(root, "*/")] + Dir[File.join(root, "templates/*/")]
```

Reply:
The code you posted does not work out-of-the-box, because `Dir[…]` would return the complete absolute paths and I couldn't subtract these from `REQUIRED_FOLDERS` so easily.

However, the more I understood the issue's actual requirements, the more I could simplify code to…
```ruby
def valid?
  REQUIRED_FOLDERS.all? { |required_folder| Dir.exist?(File.join(root, required_folder)) }
end
```
… so thanks for leading me to a better solution ✨